### DISCUSSION TITLE
Expose UI tab

### DISCUSSION THREAD

**Discussion by 19Andy89** [XKNX/knx-integration#6] [Ideas]
Hello,
The UI integration is getting better and better and that's why I would like to see a GUI tab for the "expose" feature in the future so that you can easily create lights and switches (perhaps also Binary_sensor) with HomeAssistant sensors.

That would be great.

Thanks for the awesome integration.

Best regards
Andy

**Comment by farmio** [comment]
Hi 👋! 
It was the plan to add `expose` that to UI someday. As always, if anyone is interested in contributing, feel free to reach out.
See https://github.com/XKNX/knx-frontend/discussions/51 (although at this point there is not much there).

Not really sure what you mean by 
> easily create lights and switches (perhaps also Binary_sensor) with HomeAssistant sensors

**Comment by 19Andy89** [comment]
I'm sorry for replying so late.

I thought lights and switches should be easy to implement for exposing because the GUI for creating these KNX devices already exists.

Thanks for the frontend link. I will continue to pursue the topic there.

**Reply by farmio** [reply]
Current expose (yaml) doesn't differentiate between switch, light etc. It's generic as you can just define the state or any attribute of an entity and send that value to a given GA. 
It's also only unidirectional - it only sends telegrams, but doesn't process incoming (except for read requests). 

I guess you were thinking about a bidirectional HA entity (light) to Knx binding?!

The Knx entity config GUI is built from reusable components, so parts of that could be used for other things in the panel easily.

**Reply by 19Andy89** [reply]
i were not thinking of bidirectional directly. I thought of exposing at the beginneng "simple" HA components to the KNX bus.

But you can already get lights and switching from KNX to HA with the new GUI and thats why i thought exposing HA to KNX should also  be "easy" because of the reusable components. Maybe climate is more complex as a simple binary or switch. But there are already blueprints to get for exmaple hue lights (with relative dimming and absolute dimming) to KNX with a "GUI" action. Maybe KNX could use this also as an idea.

**Comment by farmio** [comment]
An option to configure `expose` - HA entity state changes that trigger KNX telegrams - was done in https://github.com/XKNX/knx-frontend/pull/357
